jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java

```diff
@@ -58,8 +59,8 @@
     private final Executor executor;
     private final Scheduler scheduler;
     private final ManagedSelector[] _selectors;
+    private final AtomicInteger _selectorIndex = new AtomicInteger();
```

Comment:
See also comment below; I think we must use AtomicLong, which guarantees that the value remains positive for 292 years even if changed every nanosecond. AtomicInteger will wrap around in 2 seconds if changed every nanosecond, give a negative value and we'll use that negative index to access the array, which will cause an exception.

Reply:
The index will never go negative as the modulo is applied every update... however the lambda is probably a good reason for doing it differently.